jsettlers.graphics/src/jsettlers/graphics/map/MapContent.java

```diff
@@ -410,23 +417,13 @@ private void drawMain(FloatRectangle screen) {
 			}
 		}
 
-		// if (map.getConstructionPreviewBuilding() != null) {
-		// Sequence<? extends Image> sequence =
-		// ImageProvider.getInstance().getSettlerSequence(4, 5);
-		// float imageScale = Byte.MAX_VALUE / Math.max(sequence.length(), 1);
-		//
-		// ShortPoint2D underMouse =
-		// this.context.getPositionOnScreen(mousePosition.x, mousePosition.y);
-		// IHexTile tile = map.getTile(underMouse);
-		// if (tile != null) {
-		// context.beginTileContext(tile);
-		// for (ImageLink image :
-		// map.getConstructionPreviewBuilding().getImages()) {
-		// ImageProvider.getInstance().getImage(image).draw(context.getGl());
-		// }
-		// context.endTileContext();
-		// }
-		// }
+        if (placementBuilding != null) {
+            ShortPoint2D underMouse = this.context.getPositionOnScreen((float) mousePosition.getX(), (float) mousePosition.getY());
+            IMapObject mapObject = context.getMap().getMapObjectsAt(underMouse.x, underMouse.y);
+            if (mapObject != null && mapObject.getObjectType() == EMapObjectType.CONSTRUCTION_MARK) {
```

Comment:
Mind that map objects are stored in a linked list. This map object could have a child that is a Construction mark. Use a while-loop for this instead and loop through all children.